src/main/storm/mesos/MesosNimbus.java

```diff
@@ -114,94 +120,228 @@ public String getHostName(Map<String, SupervisorDetails> map, String nodeId) {
-
-        MesosSchedulerDriver driver = createMesosDriver();
-        
-        driver.start();
-    
-        LOG.info("Waiting for scheduler driver to register MesosNimbus with mesos-master and complete initialization...");
-      
-        _scheduler.waitUntilRegistered();
-      
-        LOG.info("Scheduler registration and initialization complete...");
- 
+      initialize(conf, localDir);
+
+      MesosSchedulerDriver driver = createMesosDriver();
+
+      driver.start();
+
+      LOG.info("Waiting for scheduler driver to register MesosNimbus with mesos-master and complete initialization...");
+
+      _scheduler.waitUntilRegistered();
+
+      LOG.info("Scheduler registration and initialization complete...");
+
     } catch (Exception e) {
-        LOG.error("Failed to prepare scheduler ", e);  
+      LOG.error("Failed to prepare scheduler ", e);
       throw new RuntimeException(e);
     }
   }
+
   @SuppressWarnings("unchecked")
   protected void initialize(Map conf, String localDir) throws Exception {
-      _conf  = conf;
-      _state = new LocalState(localDir);
-      _allowedHosts = listIntoSet((List<String>) conf.get(CONF_MESOS_ALLOWED_HOSTS));
-      _disallowedHosts = listIntoSet((List<String>) conf.get(CONF_MESOS_DISALLOWED_HOSTS));
-      _scheduler = new NimbusScheduler();
-      createLocalServerPort();
-      setupHttpServer();
+    _conf = new HashMap();
+    _conf.putAll(conf);
+
+    _state = new LocalState(localDir);
+    _allowedHosts = listIntoSet((List<String>) conf.get(CONF_MESOS_ALLOWED_HOSTS));
+    _disallowedHosts = listIntoSet((List<String>) conf.get(CONF_MESOS_DISALLOWED_HOSTS));
+    Boolean preferReservedResources = (Boolean) conf.get(CONF_MESOS_PREFER_RESERVED_RESOURCES);
+    if (preferReservedResources != null) {
+      _preferReservedResources = preferReservedResources;
+    }
+    _container = Optional.fromNullable((String) conf.get(CONF_MESOS_CONTAINER_DOCKER_IMAGE));
+    _scheduler = new NimbusScheduler(this);
+    createLocalServerPort();
+    setupHttpServer();
+
+    _conf.put(Config.NIMBUS_HOST, _configUrl.getHost());
+    // Generate YAML to be served up to clients
+    Path generatedConfPath = Paths.get("generated-conf");
+    if (!generatedConfPath.toFile().exists() && !generatedConfPath.toFile().mkdirs()) {
+      throw new RuntimeException("Couldn't greate generated-conf dir");
+    }
+    File generatedConf = Paths.get(generatedConfPath.toString(), "storm.yaml").toFile();
+    Yaml yaml = new Yaml();
+    FileWriter writer = new FileWriter(generatedConf);
+    yaml.dump(_conf, writer);
+  }
+
+
+  public void doRegistration(final SchedulerDriver driver, Protos.FrameworkID id) {
+    _driver = driver;
+    try {
+      _state.put(FRAMEWORK_ID, id.getValue());
+    } catch (IOException e) {
+      LOG.error("Halting process...", e);
+      Runtime.getRuntime().halt(1);
+    }
+    Number filterSeconds = Optional.fromNullable((Number) _conf.get(CONF_MESOS_OFFER_FILTER_SECONDS)).or(120);
```

Comment:
Yes, that's correct. I guess it can be thought of as 'additional filtering' for when then offers aren't required.